apps/spruce/src/pages/projectSettings/index.tsx

```diff
@@ -49,6 +51,24 @@ const ProjectSettings: React.FC = () => {
       [slugs.projectIdentifier]: string | null;
       [slugs.tab]: ProjectSettingsTabRoutes;
     }>();
+
+  const [atTop, setAtTop] = useState(true);
+  const pageWrapperRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const onScroll = throttle(() => {
+      if (!pageWrapperRef?.current) return;
+      if (pageWrapperRef?.current?.scrollTop < 40) {
+        setAtTop(true);
+      } else {
+        setAtTop(false);
+      }
+    }, 250);
+    pageWrapperRef?.current?.addEventListener("scroll", onScroll);
+    return () =>
+      pageWrapperRef?.current?.removeEventListener("scroll", onScroll);
```

Comment:
I see the following react-hooks warning
![image](https://github.com/user-attachments/assets/dac60ffe-b418-4b0b-860f-0ce910c1af8a)
Doing what it suggests seems to work and I was able to confirm the behavior